uaclient/util.py

```diff
@@ -806,3 +807,34 @@ def parse_rfc3339_date(dt_str: str) -> datetime.datetime:
     return datetime.datetime.strptime(
         dt_str_without_microseconds, "%Y-%m-%dT%H:%M:%S%z"
     )
+
+
+def validate_proxy(
+    protocol: str, proxy: Optional[str], test_url: str
+) -> Optional[str]:
```

Comment:
Validate proxy in my mind should succeed or raise exceptions if proxy values are provided. If no values are provided, a return None is in order. I'm welcome to persuasion otherwise.